editor/src/core/third-party/shared.ts

```diff
@@ -0,0 +1,20 @@
+import { getDefaultProps, PropertyControls } from 'utopia-api'
+import {
+  emptyComments,
+  JSXAttributes,
+  jsxAttributesEntry,
+  jsxAttributeValue,
+} from '../shared/element-template'
+
+export function getDefaultPropsAsAttributes(
+  propertyControls: PropertyControls | null | undefined,
+): JSXAttributes {
+  const defaultProps = getDefaultProps(propertyControls ?? {})
+  let result: JSXAttributes = []
+  for (const key of Object.keys(defaultProps)) {
+    result.push(
+      jsxAttributesEntry(key, jsxAttributeValue(defaultProps[key], emptyComments), emptyComments),
+    )
+  }
+  return result
+}
```

Comment:
```suggestion
```

Bin this all please. This function was removed in #1981 because it assumes that the property controls are correctly formed, meaning it will bring down the editor if they aren't. We have another function `getDefaultPropsFromParsedControls` that we should be using instead.

Reply:
fixed. I did not bin because it is still doing something specific to built-in tempaltes (turning an object into JSXAttributes) but it is using the safe function now

Reply:
Thanks! We're actually doing exactly this elsewhere too in [`project-components.ts`](https://github.com/concrete-utopia/utopia/blob/master/editor/src/components/shared/project-components.ts#L371-L376) so it's worth referencing the new function there then

Reply:
actually that line has been superseded in this PR, as `ComponentDescriptor` now has `componentInfo: { importsToAdd: Imports; elementToInsert: JSXElementWithoutUID }` which means no further data transformation is needed

Reply:
Even better!